src/binder_voicecall.c

```diff
@@ -1768,29 +1967,62 @@ binder_voicecall_ecclist_changed(
 {
     BinderVoiceCall* self = user_data;
     GBinderReader reader;
-    gsize count = 0;
-    const RadioEmergencyNumber* list;
     const char** en_list = NULL;
 
     /* currentEmergencyNumberList(RadioIndicationType,vec<EmergencyNumber>); */
     gbinder_reader_copy(&reader, args);
-    list = gbinder_reader_read_hidl_type_vec(&reader,
-        RadioEmergencyNumber, &count);
-
-    DBG_(self, "%u emergency number(s)", (guint) count);
-    if (count) {
-        guint i;
-
-        en_list = g_new(const char*, count + 1);
-        for (i = 0; i < count; i++) {
-            en_list[i] = list[i].number.data.str;
-            DBG("%s", en_list[i]);
+    if (self->interface_aidl == RADIO_AIDL_INTERFACE_NONE) {
+        gsize count = 0;
+        const RadioEmergencyNumber* list;
+        list = gbinder_reader_read_hidl_type_vec(&reader,
+            RadioEmergencyNumber, &count);
+
+        DBG_(self, "%u emergency number(s)", (guint) count);
+        if (count) {
+            guint i;
+
+            en_list = g_new(const char*, count + 1);
+            for (i = 0; i < count; i++) {
+                en_list[i] = list[i].number.data.str;
+                DBG("%s", en_list[i]);
+            }
+            en_list[i] = NULL;
+        }
+    } else {
+        gint32 count = 0;
+        gbinder_reader_read_int32(&reader, &count);
+
+        DBG_(self, "%u emergency number(s)", (guint) count);
+        if (count > 0) {
+            guint i;
+
+            en_list = g_new(const char*, count + 1);
+            for (i = 0; i < count; i++) {
+                gsize data_read;
+                gsize parcel_size = binder_read_parcelable_size(&reader);
+                gsize initial_size = gbinder_reader_bytes_read(&reader);
+
+                en_list[i] = gbinder_reader_read_string16(&reader);
+
+                // Ignore rest of values for now
+                data_read = gbinder_reader_bytes_read(&reader) - initial_size;
+                while (data_read < parcel_size) {
+                    gbinder_reader_read_uint32(&reader, NULL);
+                    data_read += sizeof(guint32);
+                }
+                DBG("%s", en_list[i]);
+            }
+            en_list[i] = NULL;
         }
-        en_list[i] = NULL;
     }
 
     ofono_voicecall_en_list_notify(self->vc, (char**) en_list);
-    g_free(en_list);
+
+    if (self->interface_aidl == RADIO_AIDL_INTERFACE_NONE) {
+        g_free(en_list);
+    } else {
+        g_strfreev((char**)en_list);
```

Comment:
Ehh, what? This is just asking for trouble.

Reply:
Changed the code so this kind of cast is not needed anymore.